lib/addMethod/substitute.js

```diff
@@ -24,14 +24,16 @@ function substituteString(template, params) {
     // and return the real value.
     if (template.match(/^({{([^{}]+)}}|{{{([^{}]+)}}})$/g)) {
         var key = _.trim(template.match(/(?!{)([^{}]+)(?=})/g)[0]);
-        if (!_.isUndefined(params[key]) && params[key] !== '')
-            return params[key];
+		var value = _.get(params, key, '');
+        if (value !== '') {
+			return value;
+		}
     }
 
     // If the above isn't the case, then template it with Mustache.
     var str =  Mustache.render(template, params);
 
-    return (str === '') ? undefined : str;
+    return ( str === '' ? undefined : str );
```

Comment:
can we not use _.isEmpty here?

Reply:
I rarely use `_.isEmpty`, that was a Luigi thing he liked to do in connectors.

But to answer your question, I'm going to stick to explicitly checking empty string rather than include other types. (also, I don't know if old connector code is depending on this behaviour or not).